MySQL replication lagging (#1388)
aiven/devportal: docs/products/mysql/howto/replication-lagging.rst

> Identify MySQL replication lagging

Reviewer: ```suggestion
Identify lagging MySQL replication
```

==================================

Here are some strategies to find out MySQL replication lagging on your Aiven for MySQL service.

Query locks on the standby node
-------------------------------

> Another reason you may face replication lagging is when a row or table is locked. If this happens your replication will not progress. 

Reviewer: ```suggestion
A replication might face lagging when a row or table is locked. If this happens your replication will not progress. 
```

Run the following command to check if the table level is locked::

    mysql> show processlist

> The result is a list of current processes with their statuses. If a query is causing others to lock, you should be able to identify it. The query that is causing the others queries to lock will show as waiting for another process, possibly a temporary table. The output may look like this: 

Reviewer: ```suggestion
The result is a list of current processes with their statuses. If a query is causing others to lock, you should be able to identify it. The query that is causing the others queries to lock will show as waiting for another process, possibly a temporary table. The output may look like this (check the ``State`` column): 
```

.. code::

    +-------+-----------------+------------------------------+--------+---------+-------+---------------------------------+-----------------------------------------------------------------------------------------------------+
    | Id    | User            | Host                         | db     | Command | Time  | State                           | Info                                                                                                
    |  1800 | avnadmin        | HOST_IP:53938                | teammy | Query   | 58798 | Waiting for table metadata lock | /* ApplicationName=DataGrip 2020.3.2 */ LOCK TABLES users WRITE

> If nothing shows up there, you may find them by showing the open tables::

Reviewer: ```suggestion
If no query seems being locked, you may find other table locks by showing the open tables::
```

    show open tables where In_Use > 0

Another way is to check the InnoDB status by running::

    show engine innodb status

> This will show you the locking query, and the affected rows and tables. On the output, you can check for the section ``TRANSACTION`` to find this information. There are also cases where multiple queries and load can cause replication problems.

Reviewer: ```suggestion
The above command shows the locking query, and the affected rows and tables. On the output, you can check for the section ``TRANSACTION`` to find this information. 

.. Note::
    There are also cases where multiple queries and load can cause replication problems.
```

Long running transactions
-------------------------

> Long running transactions with large binlogs can result in replication problems. You can find replication problems in this case by checking the time since the process is in progress. Once the binlog has been replicated, the standby node will be stuck processing one massive ``GTID``. If ``Read_Source_Log_Pos`` in the output is updating, this requires no further action from your side. You should leave the replication process to continue without interferring in the service.

Reviewer: there are a couple warnings here about `binlogs` and `interferring` not being part of the dictionary

Author: Thanks, I rephrased it with the formal name as **binary logs** and fix the grammar error in the other sentence.  I think it makes more sense than adding to the dict. Let me know what you think, thank you.

No disk space on the standby
----------------------------

> You can run the command ``show processlist`` to find if there is a disk space issue::

Reviewer: I would add some description here, before jumping to the code like 

"An alternative cause of lagging replication could be the missing space on the target standby database. You can check disk space issues with::"

Author: Valuable suggestion, Francesco!

.. code::

    +-------+-----------------+------------------------------+--------+---------+-------+---------------------------------+-----------------------------------------------------------------------------------------------------+
    | Id    | User            | Host                         | db     | Command | Time  | State                           | Info                                                                                                
    |  1800 | avnadmin        | HOST_IP:53938                | teammy | Query   | 58798 | Waiting for table metadata lock | /* ApplicationName=DataGrip 2020.3.2 */ LOCK TABLES users WRITE

If no query seems being locked, you may find other table locks by showing the open tables::

    show open tables where In_Use > 0

Or by checking the InnoDB status::

    show engine innodb status

The above command shows the locking query, and the affected rows and tables. On the output, you can check for the section ``TRANSACTION`` to find this information. 

.. Note::
    There are also cases where multiple queries and load can cause replication problems.

Long running transactions
-------------------------

> Long running transactions with large binary logs can result in replication problems. You can find replication problems in this case by checking the time since the process is in progress. Once the binlog has been replicated, the standby node will be stuck processing one massive ``GTID``. If ``Read_Source_Log_Pos`` in the output is updating, this requires no further action from your side. You should leave the replication process to continue without interfering in the service.

Reviewer: ```suggestion
Long running transactions with large binary logs can result in replication problems. To find these kind of replication problems check the time since the process is in progress. 

Once the binary log has been replicated, the standby node will be stuck processing one specific global transaction identifier  (``GTID``). If the ``GTID`` mentioned in the ``Read_Source_Log_Pos`` output is updating, the replication is progressing therefore requires no further action from your side.
```

Reviewer: I'm not sure where the ``GTID`` and ``Read_Source_Log_Pos`` output is, what is the command to execute?
